mm2src/coins/utxo/utxo_block_header_storage.rs

```diff
@@ -0,0 +1,162 @@
+use crate::utxo::rpc_clients::ElectrumBlockHeader;
+#[cfg(target_arch = "wasm32")]
+use crate::utxo::utxo_indexedb_block_header_storage::IndexedDBBlockHeadersStorage;
+#[cfg(not(target_arch = "wasm32"))]
+use crate::utxo::utxo_sql_block_header_storage::SqliteBlockHeadersStorage;
+use crate::utxo::UtxoBlockHeaderVerificationParams;
+use async_trait::async_trait;
+use chain::BlockHeader;
+use common::mm_ctx::MmArc;
+use common::mm_error::MmError;
+use derive_more::Display;
+use std::collections::HashMap;
+use std::fmt::{Debug, Formatter};
+
+#[derive(Debug, Deserialize, Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum BlockHeaderStorageError {
```

Comment:
As I can see, this error doesn't need to derive all serde related traits. But what's even more important: this error is already bloated and will become even bigger when you add more implementations of `BlockHeaderStorageOps`. Please use associated type instead, e.g.
https://github.com/KomodoPlatform/atomicDEX-API/blob/db95a0e345b7bc8b5686ca56553f57a9749f2c13/mm2src/coins/my_tx_history_v2.rs#L37

Reply:
> As I can see, this error doesn't need to derive all serde related traits. But what's even more important: this error is already bloated and will become even bigger when you add more implementations of `BlockHeaderStorageOps`. Please use associated type instead, e.g.
> 
> https://github.com/KomodoPlatform/atomicDEX-API/blob/db95a0e345b7bc8b5686ca56553f57a9749f2c13/mm2src/coins/my_tx_history_v2.rs#L37

Refactored as discussed on pesto.